databases/week1/homework.sql

```diff
@@ -0,0 +1,65 @@
+-- Active: 1719746560269@@127.0.0.1@3306@hyf
+
+-- 1. Find out how many tasks are in the task table
+SELECT COUNT(*) AS total_task FROM task;
+
+-- 2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*) AS total_invalid_due_date
+FROM task
+WHERE
+    due_date IS NULL;
+
+-- 3. Find all the tasks that are marked as done
+SELECT *
+FROM task
+WHERE
+    status_id = (
+        SELECT id
+        FROM status
+        WHERE
+            name = 'Done'
+    );
```

Comment:
Nice way of practicing using inner queries. But I think a JOIN would be better for this query.

Reply:
I've changed it, thank you! Is it OK that I get an additional column with the statuses because of JOIN?